src/sources/util/net/mod.rs

```diff
@@ -111,4 +167,19 @@ mod test {
         let test: Config = toml::from_str(r#"addr="systemd#3""#).unwrap();
         assert_eq!(test.addr, SocketListenAddr::SystemdFd(2));
     }
+
+    #[test]
+    fn parse_socket_listen_addr_fail() {
+        // no port specified
+        let test: Result<Config, toml::de::Error> = toml::from_str(r#"addr="127.1.2.3""#);
+        assert!(test.is_err());
+
+        // one based
```

Comment:
one based == index starts at one?

Reply:
yeah this was based on pre existing code/comments around the expectation. It's possible the nomenclature could be improved. Suggestions welcome. But I'll improve the comment here regardless.

Reply:
From the totally must be accurate [wikipedia](https://en.wikipedia.org/wiki/Zero-based_numbering) zero-based and one-based seem like the right names? I feel like I've maybe only ever hear/see zero-based which makes the other look funny 😆